### ISSUE TITLE
Kennezeichen des ZDA per OpenSSL auslesen

### ISSUE THREAD

**Issue by MartinT85** [a-sit-plus/at-registrierkassen-mustercode#374]
Hallo,
hat schon jemand Erfahrung mit dem Auslesen des ZDA Kennzeichens (z.B. 'AT1' bei A-Trust) mittels OpenSSL in C++ gemacht? Ich erhalte zwar mittels 
X509_NAME_oneline(X509_get_issuer_name(cert), NULL, 0);
die Informationen zum ZDA in der Form von:
/C=AT/O=A-Trust Ges. f. Sicherheitssysteme im elektr. Datenverkehr GmbH/OU=A-Trust Registrierkasse.CA/CN=A-Trust Registrierkasse.CA, nicht jedoch den Kenner AT1.
Danke für die Hilfestellung!

**Comment by ErichFreitag** [comment]
Sie müssen wenn VDA==A-Trust die "1" selbst zuordnen. Siehe z.B. https://github.com/a-sit-plus/at-registrierkassen-mustercode/issues/18.

**Comment by lebail** [comment]
Bei manchen Sicherheitseinrichtungen kann man dieses Kürzel auch abfragen (z.B. bei A-Trust RK Online).

In anderen Fällen muss es die Kassensoftware aber auf jeden Fall kennen. Dies gilt beispielsweise bei Nutzung einer Smartcard über APDU, wo man die Zuordnung an Hand des Kartentyps (also der ATR) selbst machen muss. Dies sollte aber kein Problem darstellen, da ja auch die restliche Interaktion mit der Karte sich von Produkt zu Produkt unterscheidet. Man muss den Typ der Karte also ohnehin ermitteln.

**Comment by TimoKunze** [comment]
Um das Kürzel zu ermitteln, lese ich per APDU das Zertifikat aus und erkenne dann anhand des Issuers die drei mir bekannten VDAs: A-Trust, GlobalTrust, Prime Sign. Das ist für mich besser nachvollziehbar als die ATR zu verwenden.

**Comment by lebail** [comment]
Kann man denn per APDU das Zertifikat auslesen, ohne vorher die ATR ausgewertet zu haben?

Ich muss zugeben, dass ich mich bis jetzt nur mit den A-Trust-Karten beschäftigt habe; aber sogar von diesen gibt es zwei verschiedene Typen, die zum Auslesen des Zertifikates unterschiedliche Kommandos erfordern. Um zu wissen, welche Karte man hat, muss man die ATR auswerten.

**Comment by jayarm** [comment]
Vielleicht hilft es :
http://labs.a-trust.at/developer/ShowSource.aspx?id=121 
[ RKInfo(...) -> ZdaId ]

**Comment by lebail** [comment]
@jayarm: Ja, so geht es. Diese Funktion ist allerdings nur verfügbar, wenn man den A-Trust-Treiber benutzt (und nicht direkt per APDU mit der Karte kommuniziert, wie es sogar von A-Trust selbst empfohlen wird).

**Comment by MartinT85** [comment]
Danke für die prompten Antworten!
Ich werde dann einfach fix "AT1" zurückliefern, da ich nur A-Trust unterstütze.
